includes/class-sensei-quiz.php

```diff
@@ -1231,6 +1231,62 @@ public function quiz_has_no_questions() {
 
 	}
 
+	/**
+	 * Check if the user is allowed to take the quiz.
+	 *
+	 * @since 3.15.0
+	 *
+	 * @param int|null $quiz_id (Optional) The quiz post ID. Defaults to the current post ID.
+	 * @param int|null $user_id (Optional) The user ID. Defaults to the current user ID.
+	 *
+	 * @return bool
+	 */
+	public function can_take_quiz( int $quiz_id = null, int $user_id = null ): bool {
```

Comment:
This method uses the same logic used in `action_buttons()`.

Reply:
Does this function need to be `public`?